sdk/core/System.ClientModel/api/System.ClientModel.net6.0.cs

```diff
@@ -49,6 +49,16 @@ public partial class OptionalClientResult<T> : System.ClientModel.ClientResult
 }
 namespace System.ClientModel.Primitives
 {
+    public partial class ApiKeyAuthenticationPolicy : System.ClientModel.Primitives.PipelinePolicy
+    {
+        internal ApiKeyAuthenticationPolicy() { }
+        public static System.ClientModel.Primitives.ApiKeyAuthenticationPolicy CreateBasicAuthorizationPolicy(System.ClientModel.ApiKeyCredential credential) { throw null; }
+        public static System.ClientModel.Primitives.ApiKeyAuthenticationPolicy CreateBearerAuthorizationPolicy(System.ClientModel.ApiKeyCredential credential) { throw null; }
+        public static System.ClientModel.Primitives.ApiKeyAuthenticationPolicy CreateHeaderApiKeyPolicy(System.ClientModel.ApiKeyCredential credential, string headerName, string? keyPrefix = null) { throw null; }
+        public static System.ClientModel.Primitives.ApiKeyAuthenticationPolicy CreateQueryApiKeyPolicy(System.ClientModel.ApiKeyCredential credential, string queryName) { throw null; }
```

Comment:
I am not sure if we want to go with "queryName" as the name. Technically, the whole section in the URI is the query. By convention, not per RFC, it's typically contains several key-value pairs separated by a delimiter. And so this maybe we name this parameter "queryKeyName"?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/41391